client/src/api/projects.js

```diff
@@ -0,0 +1,30 @@
+import httpClient from './httpClient';
+
+export const addProject = async (id, projectData) => {
+    const url = `/api/projects/add/${id}`;
+    const token = localStorage.getItem('jwtToken')
+    const authOptions = {
+        headers: {
+            Authorization: `Bearer ${token}`,
+        }
+    }
+    try {
+        const response = await httpClient.post(url, projectData, authOptions);
+        if (response.data.err) return { ...response.data }
+
+        if (typeof window !== 'undefined') {
+            let projects = [];
+
+            if (localStorage.getItem('projects')) {
+                projects = JSON.parse(localStorage.getItem('projects'));
+            }
+            projects.push(response.data);
+
+            localStorage.setItem('projects', JSON.stringify(projects))
```

Comment:
maybe we should consider making a global state in app.js instead of local? idk, I wouldn't think of storing projects in local, but I guess for caching?

Reply:
Yeah that's probably to just have the projects in a global state that initializes with a getProjects API call?